src/lifetracker/logic/StorageAdapterImpl.java

```diff
@@ -0,0 +1,27 @@
+package lifetracker.logic;
+
+import java.io.IOException;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+import lifetracker.calendar.CalendarList;
+import lifetracker.storage.Storage;
+
+public class StorageAdapterImpl implements StorageAdapter {
+
+    @Override
+    public void store(CalendarList calendar, Storage storage) throws IOException {
+        Gson gson = new GsonBuilder().setPrettyPrinting().create();
+        String s = gson.toJson(calendar);
+        storage.store(s);
+    }
+    
+    @Override
+    public CalendarList load(Storage storage) throws IOException {
+        String l = storage.load();
+        Gson gson = new Gson();
+        CalendarList calendar = gson.fromJson(l, CalendarList.class);
```

Comment:
for this line, I cannot convert the string directly to a calendarList using gson because the program says it's unable to invoke no-args constructor for calendarList.